cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/inmemory/InMemoryTable.java

```diff
@@ -36,13 +38,45 @@
   private static final long NO_TX_VERSION = 0L;
 
   private Transaction tx;
+  // name length + name of the table: handy to have one cached
+  private final byte[] nameAsTxChangePrefix;
 
+  /**
+   * To be used in tests which do not need namespaces
+   */
   public InMemoryTable(String name) {
     this(name, ConflictDetection.ROW);
   }
 
+  /**
+   * To be used in tests that do not need namespaces
+   */
```

Comment:
I think it would be better to remove this completely. Anybody who is creating a table should be namespace aware.